test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,368 @@
+    modPack = createCombo({
+        name: "modPack",
+        versionNumber: "1.0.0",
+        dependencies: ["author-mod-1.0.0"],
+        categories: ["Modpacks"]
+    });
+
+    modWithLoopDependencyA = createCombo({
+        name: "modWithLoopDependencyA",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyB-1.0.0"]
+    });
+
+    modWithLoopDependencyB = createCombo({
+        name: "modWithLoopDependencyB",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyA-2.0.0"]
+    });
+
+    modWithLoopDependencyButOneOfThemIsAModPackA = createCombo({
+        name: "modWithLoopDependencyButOneOfThemIsAModPackA",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyButOneOfThemIsAModPackB-1.0.0"],
+        categories: ["Modpacks"]
+    });
+
+    modWithLoopDependencyButOneOfThemIsAModPackB = createCombo({
+        name: "modWithLoopDependencyButOneOfThemIsAModPackB",
+        versionNumber: "1.0.0",
+        dependencies: ["author-modWithLoopDependencyButOneOfThemIsAModPackA-1.0.0"]
+    });
+};
+
+describe("DependencyUtils.getFullDependencyList", () => {
+
+    beforeAll(() => {
+        beforeSetup();
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe("Empty combo list", () => {
+        it("should return an empty list", async () => {
+            const combos: ThunderstoreCombo[] = [];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, []);
+        });
+    });
+
+    describe("Single mod without dependencies", () => {
+        it("should return single mod unchanged", async () => {
+            const combos = [mod];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, ["author-mod-1.0.0"]);
+        });
+    });
+
+
+    describe("Single mod with a dependency", () => {
+        it("should return the mod and its dependency's latest version", async () => {
+            const combos = [modWithDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-mod-2.0.0",
+                "author-modWithDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+
+    describe("A mod pack with a dependency", () => {
+        it("should return the mod pack mod and the specific version of the dependency", async () => {
+            const combos = [modPack];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modPack-1.0.0",
+                "author-mod-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod pack first and a mod with the same dependency last", () => {
+        it("should return the mod pack's specific version, " +
+            "mod's specific version, " +
+            "and the dependency's last version", async () => {
```

Comment:
Should this say "specific" instead of "last"?

---

 _🎵 To be fair 🎵_, reading what's actually going on in these tests is cumbersome, as the reader has to jump from the test to the beforeAll to see what the combos actually contain, and then to the Dexie file to see what packages are actually actually present. Ideally each test case would be self-contained i.e. the "setup phase" (currently `const combos = [something];`) would actually populate Dexie with the packages required for this test case specifically.
 
 I'm not exactly sure how much effort it would take to get the Dexie mock to support this while retaining sufficient similarity to how Dexie actually works. Also this would be a tradeoff between boilerplate code/DRY. We can discuss this in the weekly/daily meeting.